docs/content/integrations/dbt/using-dbt-with-dagster/upstream-assets.mdx

```diff
@@ -15,127 +15,140 @@ description: Dagster can orchestrate dbt alongside other technologies.
 
-## Step 2: Supply an I/O manager to the assets
+3. Then, update the model that depends on the `raw_customers` seed to instead depend on the source. Replace the contents of `model/staging/stg_customers.sql` with this:
 
-To materialize the assets, we need to tell Dagster how to handle the assets' inputs and outputs. We can do this using an [I/O manager](/concepts/io-management/io-managers).
+```sql
+with source as (
 
-In this step, we'll supply the `duckdb_io_manager` to our assets. This resource is an I/O manager that, when assets are materialized, allows:
+    {#-
+    Use source instead of seed:
+    #}
+    select * from {{ source('jaffle_shop', 'raw_customers') }}
 
-- Upstream assets (`customers_raw`, `orders_raw`) to load data into DuckDB. In this example, the `duckdb_io_manager` uses <PyObject module="dagster_duckdb_pandas" object="DuckDBPandasTypeHandler" /> to store the pandas DataFrames used in our assets as CSVs and load them into DuckDB.
-- Downstream assets to read data from DuckDB. We'll add the downstream asset in the next section.
+),
 
-In `/tutorial_template/tutorial_dbt_dagster/__init__.py`, replace its contents with the following:
+renamed as (
 
-```python
-import os
+    select
+        id as customer_id,
+        first_name,
+        last_name
 
-from dagster_dbt import DbtCliResource
-from tutorial_dbt_dagster import assets
-from tutorial_dbt_dagster.assets import DBT_PROFILES, DBT_PROJECT_PATH
+    from source
 
-from dagster_duckdb_pandas import duckdb_pandas_io_manager
+)
 
-from dagster import Definitions, load_assets_from_modules
+select * from renamed
+```
 
+4. Now let's reload our data pipeline in the UI to see what we've built:
 
-resources = {
-    "dbt": DbtCliResource(
-        project_dir=DBT_PROJECT_PATH,
-        profiles_dir=DBT_PROFILES,
-    ),
-    "io_manager": duckdb_pandas_io_manager.configured(
-        {"database": os.path.join(DBT_PROJECT_PATH, "tutorial.duckdb")}
-    ),
-}
+```shell
+cd jaffle_dagster/
 
-defs = Definitions(assets=load_assets_from_modules([assets]), resources=resources)
+DAGSTER_DBT_PARSE_PROJECT_ON_LOAD=1 dagster dev
 ```
 
----
-
-## Step 3: Materialize the assets using the Dagster UI
+## Step 3: Materialize the assets using the UI
 
 Now that you've created assets and resources, it's time to materialize the assets! Materializing an asset runs the op it contains and saves the results to persistent storage. In this tutorial, we're saving asset outputs to DuckDB.
 
 1. In the UI, click the **Reload definitions** button. This ensures that the UI picks up the changes you made in the previous steps.
 
-   At this point, the `customers_raw` and `orders_raw` assets should display above `stg_customers` and `stg_orders` as upstream dependencies:
+   At this point, the `raw_customers` asset should display above `stg_customers` as upstream dependencies:
 
    <Image
    alt="Asset graph in the Dagster UI, showing dbt models and unmaterialized assets"
```

Comment:
Screenshots have not been updated, because of the "stumped" issue I mentioned in the PR description.